src/execution_plan/execution_plan.c

```diff
@@ -330,39 +330,53 @@ static void _ExecutionPlan_PlaceApplyOps(ExecutionPlan *plan) {
 	}
 	array_free(filter_ops);
 }
+
+void ExecutionPlan_RePositionFilterOp(ExecutionPlan *plan, OpBase *search_root,
+									  const OpBase *recurse_limit, OpBase *filter) {
+	assert(filter->type == OPType_FILTER);
+	rax *references = FilterTree_CollectModified(((OpFilter *)filter)->filterTree);
+	OpBase *op;
+	if(raxSize(references) > 0) {
+		/* Scan execution plan, locate the earliest position where all
+		 * references been resolved. */
+		op = ExecutionPlan_LocateReferences(search_root, recurse_limit, references);
+	} else {
+		/* The filter tree does not contain references, like:
+		 * WHERE 1=1
+		 * TODO This logic is inadequate. For now, we'll place the op
+		 * directly below the first projection (hopefully there is one!). */
+		op = plan->root;
+		while(op->childCount > 0 && op->type != OPType_PROJECT && op->type != OPType_AGGREGATE) {
+			op = op->children[0];
+		}
+	}
+	assert(op);
+	// In case this is a pre-existing filter (this function is not called out from ExecutionPlan_PlaceFilterOps)
```

Comment:
"this function is not called out from ExecutionPlan_PlaceFilterOps" ?

Reply:
you can call this function not only from ExecutionPlan_PlaceFilterOps)